streamly.cabal

```diff
@@ -691,13 +689,11 @@ library
 
                      -- XXX to be removed
                      , transformers-base >= 0.4   && < 0.5
-                     , primitive         >= 0.5.4 && < 0.8
```

Comment:
Does it build with the `dev` flag? This dependency is remove unconditionally, isn't it needed for building with SmallArray in `dev` flag case?